src/pages/Leaderboard.tsx

```diff
@@ -0,0 +1,248 @@
+import {
+  Box,
+  Flex,
+  Heading,
+  Input,
+  InputGroup,
+  Table,
+  TableCaption,
+  Tbody,
+  Td,
+  Text,
+  Th,
+  Thead,
+  Tr,
+} from "@chakra-ui/react";
+import { useGetLeaderboard } from "../hooks/useGetLeaderboard";
+import { useMemo, useState } from "react";
+import { ChevronDownIcon, ChevronUpIcon } from "@chakra-ui/icons";
+import { Header } from "../components";
+import { useAccount } from "wagmi";
+import { isAddress } from "ethers/lib/utils.js";
+
+export default function Leaderboard() {
+  const { data, loading } = useGetLeaderboard();
+  const [sortOption, setSortOption] = useState(["ranking", true]);
+  const { address } = useAccount();
+  const [filterUser, setFilterUser] = useState("");
+  const userPerformance = data?.leaderboard.filter(
+    (user) => user.address.toLowerCase() === address?.toLowerCase(),
+  );
+
+  const isFilterValid = useMemo(
+    () => !filterUser || isAddress(filterUser),
+    [filterUser],
+  );
+
+  const filteredList = useMemo(() => {
+    if (filterUser) {
+      return (
+        data?.leaderboard.filter((item) =>
+          item.address.toLowerCase().includes(filterUser.toLowerCase()),
+        ) || []
+      );
+    }
+
+    return data?.leaderboard || [];
+  }, [data, filterUser]);
+
+  const topTraders = useMemo(() => {
+    return data?.leaderboard.filter((_, index) => index < 10);
+  }, [data]);
+
+  if (loading) {
+    return <Text>Loading...</Text>;
+  }
+
+  return (
+    <>
+      <Header />
+      <Flex flexDir="column" alignItems="center" p="10">
+        <Heading textAlign="start" w="100%" mt="12">
+          Trading Competition
+        </Heading>
+
+        <Box
```

Comment:
i think it would be better to render the header in loading so:
{loading ? <Text>Loading...</Text> :  <Flex flexDir="column" alignItems="center" p="10">...}

Reply:
I also suggest showing an skeleton